src/MergeContext.js

```diff
@@ -883,11 +837,6 @@ class MergeFinalizer extends MergeContext {
         if (!(await this._messageIsFresh()))
             return await this._cleanupMergeFailed(true, this._labelCleanStaged);
 
-        this._approval = await this._checkApproval();
-        this._log("checkApproval: " + this._approval);
-        await this._setApprovalStatus(this._prHeadSha());
-        await this._setApprovalStatus(this._tagSha);
-
         if (this._prInProgress()) {
```

Comment:
Some conditions, like this one (if it is interpreted without peeking at its current implementation) and _prOpen() are both pre- and post-conditions. Have you tried grouping them into a dedicated _checkTimelessConditions() method that the other two _check*conditions() methods would call?

Reply:
I do not understand the parts of your your response talking about _prInProgress() -- AFAICT that method checks the `WIP:` prefix. Checking that PR title prefix has nothing to do with tracking already merged PRs. While it is currently recorded as a part of the title, the prefix should be treated like a label. I suggest s/InProgress/WorkInProgress/ or even s/prInProgress/wipPr/ to reduce clashes between "PR merging progress" and "work-in-progress PR".